### PR TITLE
Fix 0.5 deprecations and add the tests badge

### PR DESCRIPTION
**Pull request by ararslan**

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -15,8 +15,7 @@ test_infile = @__FILE__
 test_compressed = joinpath(tmp, "runtests.jl.gz")
 test_empty = joinpath(tmp, "empty.jl.gz")
 
-@windows_only gunzip="gunzip.exe"
-@unix_only    gunzip="gunzip"
+gunzip = "gunzip" * (is_windows() ? ".exe" : "")
```
Comment (kmsquire) [inline test/runtests.jl]: Clever.

Not a big deal if you choose not to, but I think this is clearer written closer to the original way.

Comment (ararslan) [inline test/runtests.jl]: Okay, I'll change it. Thanks for the feedback!

**Comment by ararslan** [comment]
Getting the following error on 0.3:
```
ERROR: type: readall: in apply, expected Function, got Type{Union(UTF8String,ASCIIString)}
 in readall at C:\Users\appveyor\.julia\v0.3\GZip\src\GZip.jl:419
 in gzopen at C:\Users\appveyor\.julia\v0.3\GZip\src\GZip.jl:270
```
It seems like it's trying to use the `gzopen(f::Function, ...)` method when passed a string. I don't have 0.3 locally to test, but is that union type not recognized as an `AbstractString`?

File: src/GZip.jl [hunk 2]
```diff
@@ -414,7 +416,7 @@ function readall(s::GZipStream, bufsize::Int)
             if length(buf) > len
                 resize!(buf, len)
             end
-            return bytestring(buf)
+            return UTF8String(copy(buf))
```
Comment (tkelman) [inline src/GZip.jl]: I think this is changing the return type? Is there a `Compat.bytestring` that maintains the 0.4 return type when run on that version?

Comment (ararslan) [inline src/GZip.jl]: I think I misunderstood what I had read in Compat. Looking again, I think it should be `@compat String(copy(v))`?

**Comment by quinnj** [comment]
Has appveyor ever worked on this package? Looks like it's complaining about `gunzip.exe` not existing.

**Comment by tkelman** [comment]
No I don't think it ever has. BTW we should move this package somewhere like JuliaIO for CI queue purposes. Anyone have objections to doing that?

**Comment by quinnj** [comment]
What we really need to do is merge Libz.jl and this package so we have proper file-based and stream-based gzipping all in one place. It's been on my todo list for a while, maybe I just need to hunker down and do it.

**Comment by quinnj** [comment]
But yes, in the mean time, we can certainly move this to JuliaIO

**Comment by tkelman** [comment]
Isn't Zlib.jl the newest replacement?

**Comment by quinnj** [comment]
No, Libz.jl is newer than Zlib. Libz.jl has a proper stream-based implementation, but doesn't have the file-based interface.

**Comment by ararslan** [comment]
+1 for moving this to JuliaIO.

Wow, the Travis build has been in queue for about 10 hours or so now.

**Comment by tkelman** [comment]
Travis on JuliaLang is all kinds of broken this week.

**Comment by ararslan** [comment]
Whoa, did this move within the last 10 minutes or am I just super oblivious?

**Comment by tkelman** [comment]
I did it about half an hour ago, yeah

**Comment by ararslan** [comment]
Wow, looks like it's still queued here too. Is Travis borken across the board?

**Comment by tkelman** [comment]
might be. or it could be confused on which queue to count this under since it just moved?

**Comment by ararslan** [comment]
Perhaps. The official Travis CI status says that there's been a partial outage for OS X for the past couple of hours. Looks like Linux is running and passing here.

AppVeyor also seems to have magically disappeared in the move.

**Comment by tkelman** [comment]
whose account was the running on? probably need to reauthorize it.

**Comment by ararslan** [comment]
I mean the badge and everything is gone. Am I going crazy?

I think it was running on @yuyichao's account when it was on JuliaLang.

**Comment by yuyichao** [comment]
I updated the settings and manually triggered a rebuild. Let me know if automatic build still doesn't work (in which case I'll probably ask @tkelman for help =) ).

**Comment by ararslan** [comment]
Thanks so much, @yuyichao! You rock!

**Comment by ararslan** [comment]
Cool, so Travis finally finished and things are looking good on OS X and Linux. AppVeyor is still failing for the reason that @quinnj mentioned. I can have a go at getting it to work on AppVeyor but that's kind of outside of the scope of this PR.

File: test/runtests.jl [hunk 3]
```diff
@@ -15,8 +15,11 @@ test_infile = @__FILE__
 test_compressed = joinpath(tmp, "runtests.jl.gz")
 test_empty = joinpath(tmp, "empty.jl.gz")
 
-@windows_only gunzip="gunzip.exe"
-@unix_only    gunzip="gunzip"
+if is_windows()
+    gunzip = "gunzip.exe"
+elseif is_unix()
+    gunzip = "gunzip"
+end
```
Comment (kmsquire) [inline test/runtests.jl]: I just merged this, but this should probably be made exhaustive--either by removing the second test, or by having a third else which throws an error (i.e., for the potential case of Julia being ported to some non-unix, non-windows system).

Comment (kmsquire) [inline test/runtests.jl]: (This problem existed in the original as well, of course!)

Comment (ararslan) [inline test/runtests.jl]: Okay, I can submit a separate PR for that. Thanks!

**Pull request merged**